api/sys_plugins_runtimes.go

```diff
@@ -0,0 +1,191 @@
+		return err
+	}
+
+	resp, err := c.c.rawRequestWithContext(ctx, req)
+	if err == nil {
+		defer resp.Body.Close()
+	}
+	return err
+}
+
+// DeregisterPluginRuntimeInput is used as input to the DeregisterPluginRuntime function.
+type DeregisterPluginRuntimeInput struct {
+	// Name is the name of the plugin runtime. Required.
+	Name string `json:"-"`
+
+	// Type of the plugin. Required.
+	Type PluginRuntimeType `json:"type"`
+}
+
+// DeregisterPluginRuntime removes the plugin with the given name from the plugin
+// catalog.
+func (c *Sys) DeregisterPluginRuntime(ctx context.Context, i *DeregisterPluginRuntimeInput) error {
+	ctx, cancelFunc := c.c.withConfiguredTimeout(ctx)
+	defer cancelFunc()
+
+	path := pluginRuntimeCatalogPathByType(i.Type, i.Name)
+	req := c.c.NewRequest(http.MethodDelete, path)
+	resp, err := c.c.rawRequestWithContext(ctx, req)
+	if err == nil {
+		defer resp.Body.Close()
+	}
+	return err
+}
+
+type PluginRuntimeDetails struct {
+	Type         string `json:"type"`
+	Name         string `json:"name"`
+	OCIRuntime   string `json:"oci_runtime"`
+	CgroupParent string `json:"cgroup_parent"`
+	CPU          int64  `json:"cpu"`
+	Memory       int64  `json:"memory"`
+}
+
+// ListPluginRuntimesInput is used as input to the ListPluginRuntimes function.
+type ListPluginRuntimesInput struct {
+	// Type of the plugin. Required.
+	Type PluginRuntimeType `json:"type"`
+}
+
+// ListPluginRuntimesResponse is the response from the ListPluginRuntimes call.
+type ListPluginRuntimesResponse struct {
+	// RuntimesByType is the list of plugin runtimes by type.
+	Runtimes []PluginRuntimeDetails `json:"runtimes"`
+}
+
+// ListPluginRuntimes lists all plugin runtimes in the catalog and returns their names as a
+// list of strings.
+func (c *Sys) ListPluginRuntimes(ctx context.Context, i *ListPluginRuntimesInput) (*ListPluginRuntimesResponse, error) {
+	ctx, cancelFunc := c.c.withConfiguredTimeout(ctx)
+	defer cancelFunc()
+
+	if i != nil && i.Type == PluginRuntimeTypeUnsupported {
+		return nil, fmt.Errorf("%q is not a supported runtime type", i.Type.String())
+	}
+
+	resp, err := c.c.rawRequestWithContext(ctx, c.c.NewRequest(http.MethodGet, "/v1/sys/plugins/runtimes/catalog"))
+	if err != nil && resp == nil {
+		return nil, err
+	}
+	if resp == nil {
+		return nil, nil
+	}
+	defer resp.Body.Close()
+
+	secret, err := ParseSecret(resp.Body)
+	if err != nil {
+		return nil, err
+	}
+	if secret == nil || secret.Data == nil {
+		return nil, errors.New("data from server response is empty")
+	}
+	if _, ok := secret.Data["runtimes"]; !ok {
+		return nil, fmt.Errorf("data from server response does not contain runtimes")
+	}
+
+	runtimesJSON, err := json.Marshal(secret.Data["runtimes"])
+	if err != nil {
+		return nil, fmt.Errorf("unable to marshal runtimes")
+	}
+
+	var runtimes []PluginRuntimeDetails
+	if err = json.Unmarshal(runtimesJSON, &runtimes); err != nil {
+		return nil, fmt.Errorf("unable to parse runtimes")
+	}
```

Comment:
I had a bit of struggle parsing the Data back to runtimes after changing to store the returned runtimes in Data as `[]map[string]any` https://github.com/hashicorp/vault/blob/9e59d843b4a0ce841e74eff47348fbcc8e6f4744/vault/logical_system.go#L835-L871  like in the example Response below.
```
{
    "request_id": "e93d3f93-8e4f-8443-a803-f1c97c123456",
    "data": {
        "runtimes": [
			{
				"name": "gvisor",
				"type": "container",
				"oci_runtime": "runsc",
				"cgroup_parent": "/cpulimit/",
				"cpu": 1,
				"memory": 10000
			},
			{
				"name": "foo",
				"type": "container",
				"oci_runtime": "otherociruntime",
				"cgroup_parent": "/memorylimit/",
				"cpu": 2,
				"memory": 20000
			}
		]
    },
    "warnings": null,
    "auth": null
}
```

I attempted use [`mapstructure.Decode`](https://github.com/hashicorp/vault/blob/8ebbdfba3fd1d30871a01daff89655646bfacc20/api/sys_plugins_runtimes.go#L165-L180) different ways to parse Response.Data["runtimes"] 
but `Test*List*` sys_plugins_runtimes_test.go couldn't pass as`OCIRuntime:"", CgroupParent:""` weren't parsed correctly.
```
=== RUN   TestListPluginRuntimeTyped/container
    sys_plugins_runtimes_test.go:135: expected: &api.ListPluginRuntimesResponse{Runtimes:[]api.PluginRuntimeDetails{api.PluginRuntimeDetails{Type:"container", Name:"gvisor", OCIRuntime:"runsc", CgroupParent:"/cpulimit/", CPU:1, Memory:10000}}}
        got: &api.ListPluginRuntimesResponse{Runtimes:[]api.PluginRuntimeDetails{api.PluginRuntimeDetails{Type:"container", Name:"gvisor", OCIRuntime:"", CgroupParent:"", CPU:1, Memory:10000}}}
=== RUN   TestListPluginRuntimeTyped/unsupported
--- FAIL: TestListPluginRuntimeTyped (0.01s)

=== RUN   TestListPluginRuntimeUntyped/#00
    sys_plugins_runtimes_test.go:197: expected: &api.ListPluginRuntimesResponse{Runtimes:[]api.PluginRuntimeDetails{api.PluginRuntimeDetails{Type:"container", Name:"gvisor", OCIRuntime:"runsc", CgroupParent:"/cpulimit/", CPU:1, Memory:10000}, api.PluginRuntimeDetails{Type:"container", Name:"foo", OCIRuntime:"otherociruntime", CgroupParent:"/memorylimit/", CPU:2, Memory:20000}, api.PluginRuntimeDetails{Type:"container", Name:"bar", OCIRuntime:"otherociruntime", CgroupParent:"/cpulimit/", CPU:3, Memory:30000}}}
        got: &api.ListPluginRuntimesResponse{Runtimes:[]api.PluginRuntimeDetails{api.PluginRuntimeDetails{Type:"container", Name:"gvisor", OCIRuntime:"", CgroupParent:"", CPU:1, Memory:10000}, api.PluginRuntimeDetails{Type:"container", Name:"foo", OCIRuntime:"", CgroupParent:"", CPU:2, Memory:20000}, api.PluginRuntimeDetails{Type:"container", Name:"bar", OCIRuntime:"", CgroupParent:"", CPU:3, Memory:30000}}}
--- FAIL: TestListPluginRuntimeUntyped (0.01s)
```

Reply:
After `mapstructure.Decode` attemps, I resorted to use `json.Marshal` and `json.Unmarshal` to put Data["runtimes"] in JSON bytes and let the library do the work, and the tests pass now. Admittedly, I feel this is a bit weird and hacky way. I'm open to any thoughts and tips that you have.